src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,33 @@
+(ns goose.brokers.redis.batch
+  (:require
+    [goose.brokers.redis.commands :as redis-cmds]
+    [goose.retry :as retry]))
+
+(def ^:no-doc batch-prefix "goose/batch")
+
+(defn callback-set [batch-id] (str batch-prefix "/" batch-id))
+(defn enqueued-set [batch-id] (str batch-prefix "/" batch-id ":enqueued"))
+(defn successful-set [batch-id] (str batch-prefix "/" batch-id ":successful"))
+(defn failed-set [batch-id] (str batch-prefix "/" batch-id ":failed"))
+
+(defn update-stats [next]
+  (fn [{:keys [redis-conn]
+        :as   opts}
+       {:keys [id, batch-id]} job]
+    (if (contains? job :batch-id)
+      (try
+        (let [response (next opts job)
+              _ (redis-cmds/move-in-set redis-conn
```

Comment:
question: Does this sexp need to be inside the let block? It could be placed just above response (the last sexp)